crates/polars-core/src/serde/series.rs

```diff
@@ -204,8 +204,11 @@ impl<'de> Deserialize<'de> for Series {
                     },
                     DataType::List(_) => {
                         let values: Vec<Option<Series>> = map.next_value()?;
+                        // retain dtype information in case of no actual values
                         if values.is_empty() {
                             Ok(Series::new_empty(&name, &dtype))
+                        } else if values.iter().all(|s| s.is_none()) {
```

Comment:
We should use the `AnonymousListBuilder here`. It accepts the `inner_dtype` of the `List` as argument upon construction and will ensure we don't do a double pass on the data.